### PR TITLE
Add API testdata population instructions

### PR DESCRIPTION
**Pull request by liggitt**
/kind cleanup
/cc @dims

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/api/testdata/README.md [hunk 1]
```diff
@@ -2,6 +2,21 @@
...
+
+For example, to capture compatibility data for `v1.20.0`:
+
+```sh
+export VERSION=v1.20.0
+git checkout ${VERSION}
+cp -fr staging/src/k8s.io/api/testdata/{HEAD,${VERSION}}
+git checkout -b ${VERSION}-api-testdata master
+git add .
+git commit -m "Add ${VERSION} API testdata"
```
Comment (krmayankk) [inline staging/src/k8s.io/api/testdata/README.md:17]: what is this used for ?

Comment (liggitt) [reply staging/src/k8s.io/api/testdata/README.md:17]: is the question about the entire testdata folder, or this specific line? this testdata is used to ensure we are backwards compatible with data serialized from the API definitions in previous releases.

**Pull request merged**